archive_envoy_release.sh

```diff
@@ -29,7 +29,7 @@
 #
 # Notes:
```

Comment:
not relevant to this PR though, can we put `archive_envoy_release.sh` under bin/ as well? And the word `archive` makes me really confused because that is used here for `doing tar` vs `uploading to github release assert` in `archive_release_version`..

so I would suggest rename this shell to `tar_${}_release.sh` or soemthing?

Reply:
it is the only thing that has Envoy trademark, the rest of the scripts are generic.. that's why

Reply:
not only tar but also gzip though 😄

Reply:
i see.. but maybe I think we need more distinguished names,,

Reply:
on the other topic, the name was more about the result not the impl. ex "archive_envoy_release" results in the files that correspond exactly to a release in the archive-envoy repo. To do this, there are many steps and tar is one thing, but also the json is generated

Reply:
I think the main docs on that file drifted, so I'll update it.

Reply:
I'll fix the filename too, as you are right actually. I still hesitate about adding Envoy® to more files, so prefer to keep that by itself for now

Reply:
ok I think it is better now. you were totally right on tar, btw